generators/GitHubIssueGenerator.py

```diff
@@ -215,18 +235,119 @@ def generate_github_issue_from_args(args):
+        # Get a view of just the metadata to extract the list of squads
+        _metadata = []
+        for f in _failures:
+            _metadata.append(f['metadata'])
+        # What are the unique squads in that metadata?
+        _squads = list(set(itertools.chain(*map(lambda r: r.get('squad(s)', "Unlabelled"), _metadata))))
+        # Iterate over all squads and open an issue for that squad if one doesn't already exist for this set of failures
+        for squad in _squads:
+            _squad_issue_set = list(filter(lambda a: a['metadata']['squad(s)'][0] == squad, _failures))
+            # Need to flatten issue set down to name, testsuite, squad, pri, sev
+            _flat_issue_set = []
+            _highest_sev=len(GitHubIssueGenerator.tag_mappings)
+            _highest_pri=len(GitHubIssueGenerator.tag_mappings)
+            for issue in _squad_issue_set:
+                new_issue = {"name":issue['name'],"testsuite":issue['testsuite']}
+                sev = GitHubIssueGenerator.tag_mappings.get(issue['metadata']['severity'].lower(), None)
+                pri = GitHubIssueGenerator.tag_mappings.get(issue['metadata']['priority'].lower(), None)
+                _highest_sev = GitHubIssueGenerator.severities.index(sev) if sev in GitHubIssueGenerator.severities and GitHubIssueGenerator.severities.index(sev) < _highest_sev else _highest_sev
+                _highest_pri = GitHubIssueGenerator.priorities.index(pri) if pri in GitHubIssueGenerator.priorities and GitHubIssueGenerator.priorities.index(pri) < _highest_pri else _highest_pri
+                _flat_issue_set.append(new_issue)
+            if _highest_sev == len(GitHubIssueGenerator.tag_mappings):
+                # A severity wasn't found, for some reason - assign highest
+                _highest_sev = 0
+            if _highest_pri == len(GitHubIssueGenerator.tag_mappings):
+                # A priority wasn't found, for some reason - assign highest
+                _highest_pri = 1 # 0 would be blocker, that's a little aggressive
+            if len(_flat_issue_set) > 0:
+                # Iterate through db, see if issue exists already
+                dup = db_utils.payload_exists(_flat_issue_set)
+                if dup == None:
+                    # Get a local copy of tags they want us to add
+                    _tags = self.tags.copy()
+                    _tags.append(GitHubIssueGenerator.severities[_highest_sev])
+                    _tags.append(GitHubIssueGenerator.priorities[_highest_pri])
+                    _tags.append("squad:{}".format(squad))
+                    github_id = self.open_github_issue_per_squad(_tags)
+                    if github_id == None:
+                        github_id = "seed{}".format(randrange(10000,99999))
+                    print("Unique issue, adding github id {} priorty '{}' and severity '{}' for squad:{}.".format(github_id,GitHubIssueGenerator.severities[_highest_sev],GitHubIssueGenerator.priorities[_highest_pri],squad))
```

Comment:
I won't add this for every print occurrence, but at the end of the run, GitHubIssueGenerator prints the [opened GitHub issue](https://github.com/open-cluster-management/canary-reporting/blob/main/generators/GitHubIssueGenerator.py#L253) which is used to [feed into the other canary report generators](https://github.com/open-cluster-management/canary-scripts/blob/7b46cc6546837ae3ee14a45830a5c4ac445f45de/failure.sh#L26).

Reply:
In short - the "happy path" through the code should _only_ output the URL.  That being said I'm open to other options as well!

Reply:
We will potentially have multiple, or zero defects opened even when things went badly (because all might be dups, for example).  I am satisfied with how the message gets sent to slack, for example - it says what happened, which makes sense.  Where else is this (what used to be one and only one) URL used?

Reply:
Understood, approval incoming!